packages/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -200,9 +200,16 @@ class GraphQLRequestFactory {
 
     // e.g. { 'name': { 'eq': 'foo }}
     if (queryPredicate is QueryPredicateOperation) {
-      // check for the IDs where fieldName set to e.g. "blog.id" and convert to "id"
-      final isId = queryPredicate.field == '${schema.name.toLowerCase()}.id';
-      final fieldName = isId ? idFieldName : queryPredicate.field;
+      final associatedTargetName =
+          schema.fields?[queryPredicate.field]?.association?.targetName;
+      String fieldName = queryPredicate.field;
+      if (queryPredicate.field == '${schema.name.toLowerCase()}.$idFieldName') {
```

Comment:
I just realized, this should be "camel-cased" schema name, not the lower-cased one: https://github.com/aws-amplify/amplify-codegen/blob/8baa1d91db83911c6b7fbc2d1c4601a525f37ba1/packages/appsync-modelgen-plugin/src/visitors/appsync-dart-visitor.ts#L752

Reply:
good catch! updated to only lowercase first letter like the transformer